src/rpp/rpp/operators/details/combining_strategy.hpp

```diff
@@ -0,0 +1,125 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/disposables/composite_disposable.hpp>
+#include <rpp/operators/details/strategy.hpp>
+#include <rpp/operators/details/utils.hpp>
+#include <rpp/schedulers/current_thread.hpp>
+
+#include <memory>
+
+namespace rpp::operators::details
+{
+template<rpp::constraint::observer Observer, typename TSelector, rpp::constraint::decayed_type... Args>
+class combining_disposable : public composite_disposable
+{
+public:
+    explicit combining_disposable(Observer&& observer, const TSelector& selector)
+        : m_observer_with_mutex{std::move(observer)}
+        , m_selector{selector}
+    {
+    }
+
+    pointer_under_lock<Observer> get_observer_under_lock() { return pointer_under_lock{m_observer_with_mutex}; }
+
+    const TSelector& get_selector() const { return m_selector; }
+
+    bool decrement_on_completed()
+    {
+        // just need atomicity, not guarding anything
+        return m_on_completed_needed.fetch_sub(1, std::memory_order::seq_cst) == 1;
+    }
+
+private:
+    value_with_mutex<Observer>      m_observer_with_mutex{};
+    RPP_NO_UNIQUE_ADDRESS TSelector m_selector;
```

Comment:
Not sure if this one should be in base class. Actually it doesn't need it at all

Reply:
done